### ISSUE TITLE
Cannot log into backend in PHP 7.0

### ISSUE THREAD

**Issue by timokleemann** [symphonycms/symphony-2#2869]
I recently upgraded my (shared) webspace at HostEurope from PHP 5.6 to PHP 7.0.

For some reason, I can no longer log into my site. When I enter my user name and password, the page gets reloaded but nothing happens, not even an error message gets thrown. 

This happened first in Symphony 2.6.X, then I upgraded to the latest version (2.7.7) because I thought that would solve it. But it didn't.

I've already cleared all my browser cashes and cookies and tried various browsers, but to no avail. 

I've also tried PHP 7.1 and 7.2 but that doesn't change anything.

I am running more than a dozen Symphony websites. This is the first one on PHP 7.

Can anybody help?

Affected Symphony version(s) : 2.6.X to 2.7.7
PHP version(s) : 7.X
OS(es) : Mac OS 10.13.6
Browsers: Safari, Chrome, Firefox

**Comment by twiro** [comment]
PHP 7.X support started with Symphony 2.7.X, so you won't be able to run Symphony 2.6.X on PHP 7.X.

Your scenario sounds like you only updated the Symphony core and not the extensions (as you couldn't access the backend after having already caused the backend crash after upgrading to PHP 7.X), so I guess you are using one ore more extensions that aren't compatible with PHP 7.X.

Your Host Europe PHP error logs should be able to tell you which extension(s) might cause trouble.

I also run lots of Symphony websites on Host Europe and sucessfully updated most of them to Symphony 2.7.7 + PHP 7.1, but this process involved carefully checking all extensions for PHP 7.X-support and replacing them (or updating them by myself) if they're outdated or no more supported.

If you're able to post a list of your installed extensions and their version numbers we might be able to give you some more useful hints which ones might cause problems…

**Comment by nitriques** [comment]
Yeah please post an error message from either `error_log` or `manifest/logs/main`. Thanks

**Comment by twiro** [comment]
_> I just double checked all my extensions as you recommended. They're all up to date._

Thanks – looks like a pretty basic (and therefore rather unproblematic) setup.

_> The only errors I could find in Host Europe's error logs are these two:_

As Host Europe has separate error logs for each PHP-version… are these the errors shown in Host Europes error log for PHP 7.0? And they are the only ones?

If so this doesn't look like any of the extensions-compatibility-issues I faced while updating – and if it's not an extension that's not ready for PHP7 I have to admit that I have no idea what might cause your backend-login-troubles…

**Comment by timokleemann** [comment]
Yes, I checked the error logs for all PHP versions (5.6, 7.0, 7.1 and 7.2). I could only find the two errors above in there.

**Comment by nitriques** [comment]
@timokleemann What about `main` under `manifest/logs` ? Nothing in there ?

Also, try to

1) switch to PHP 7.1
2) delete all your cookies
3) delete all rows in the `sym_sessions` tables
4) log in

If log in still does not work:

1) Try to see if the xsrf token on the login page changes each time you refresh the page (it is in a hidden field).
1) If so, please tell us !

**Comment by timokleemann** [comment]
Hi,

I just realized that there is **no** hidden field for any `xsrf token` in my login form! It's just not there!

(Since I run a number of other Symphony websites, I checked some of those too, and all of them **do** have an `xsrf token` in their login forms.)

The only relevant error I could find in `manifest/log/main` is: 

`Warning: GenericErrorHandler 2: session_decode(): Session is not active. You cannot decode session data on line 198 of file /is/htdocs/wp1138379_3WGLF7BD3N/www/symphony/lib/core/class.session.php`

The others are probably not relevant:

`User Notice: FrontendPageNotFoundException 1024 - The page you requested, <code>/apple-touch-icon-precomposed.png/</code>, does not exist. on line 328 of /is/htdocs/wp1138379_3WGLF7BD3N/www/symphony/lib/toolkit/class.frontendpage.php`

`User Notice: FrontendPageNotFoundException 1024 - The page you requested, <code>/apple-touch-icon-precomposed.png/</code>, does not exist. on line 328 of /is/htdocs/wp1105399_3WHLF6BD4N/www/symphony/lib/toolkit/class.frontendpage.php`

**Comment by nitriques** [comment]
> I just realized that there is no hidden field for any xsrf token in my login form! It's just not there!

Is is enabled in config.php ?

> Session is not active. You cannot decode session 

We have [a fix for PHP 7.0](https://github.com/symphonycms/symphony-2/commit/67ad067ce0b338d254a88aa2816a59267e991fd9) which is in 2.7.7. Did you make sure it is up to date ?

**Comment by timokleemann** [comment]
No, it wasn't enabled. I just enabled it and got the `missing cross site request token` error.

I fixed it by applying the [fix for PHP 7.0](https://github.com/symphonycms/symphony-2/commit/67ad067ce0b338d254a88aa2816a59267e991fd9) which you mentioned.

(Sorry, but I was under the impression that it's already part of Symphony 2.7.7)

It's all fixed now. I am closing this. Thanks a lot for your help.
